core/src/cache_block_meta_service.rs

```diff
@@ -59,10 +59,13 @@ impl CacheBlockTimeService {
         Self { thread_hdl }
     }
 
-    fn cache_block_time(bank: Arc<Bank>, blockstore: &Arc<Blockstore>) {
+    fn cache_block_meta(bank: Arc<Bank>, blockstore: &Arc<Blockstore>) {
         if let Err(e) = blockstore.cache_block_time(bank.slot(), bank.clock().unix_timestamp) {
             error!("cache_block_time failed: slot {:?} {:?}", bank.slot(), e);
         }
+        if let Err(e) = blockstore.cache_block_height(bank.slot(), bank.block_height()) {
```

Comment:
@CriesofCarrots just summarizing our discussion, because this is async with replay, there's the possibility that somebody queries a block and the status/height is not yet written, in which case we'll return `None` for those fields.

Should we document this behavior?

Reply:
Good call. I'll add a comment in `Blockstore::get_complete_block` that speaks to block time and block height.